workers/unity/Assets/Gdk/TestUtils/Tests/Editmode/HybridGdkSystemTests.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using JetBrains.Annotations;
+using NUnit.Framework;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.TestUtils.EditmodeTests
+{
+    public class HybridSystemTests
+    {
+        private struct TestPreparation : IComponentData
+        {
+            public int Value;
+        }
+
+        [UsedImplicitly]
+        [DisableAutoCreation]
+        private class ExampleHybridSystem : ComponentSystem
+        {
+#pragma warning disable 649
+            private struct PreparationData
+            {
+                public int Length;
+
+                // The ComponentArray<> here makes this a hybrid system.
+                public ComponentArray<Rigidbody> GameObjectRigidBody;
+
+                public ComponentDataArray<TestPreparation> PreparationStruct;
+            }
+
+            [Inject] private PreparationData testDataToPrepare;
+#pragma warning restore 649
+
+            public void TestInjection(World world)
+            {
+                OnBeforeCreateManagerInternal(world, 10);
+            }
+
+            public void ManualDispose()
+            {
+                OnBeforeDestroyManagerInternal();
+                OnDestroyManager();
+                OnAfterDestroyManagerInternal();
+            }
+
+            protected override void OnUpdate()
+            {
+                for (var i = 0; i < testDataToPrepare.Length; ++i)
+                {
+                    var preparation = testDataToPrepare.PreparationStruct[i];
+
+                    preparation.Value = 1;
+
+                    testDataToPrepare.PreparationStruct[i] = preparation;
+                }
+            }
+        }
+
+        [TestFixture]
+        private class FixtureNotImplementingHybridTestBase
+        {
+            // If this test fails, then we can get rid of HybridGdkSystemTestBase!
+            [Test]
+            public static void
+                HybridSystems_will_result_in_errors_when_created_if_fixture_does_not_extend_HybridGdkSystemTestBase()
```

Comment:
Now that's a name. :P

Reply:
`HybridSystems_will_throw_unless_fixture_extends_HybridGdkSystemTestBase`

Reply:
Or `HybridSystems_fixture_must_extend_HybridGdkSystemTestBase`?

Reply:
I don't think this is actually a problem if it's long as it's meant to be a read in a log rather than, used in code.
However if you have to scroll right to read it then it's less than ideal :P